app/src/main/java/com/example/photofy/fragments/CameraFragment.java

```diff
@@ -0,0 +1,154 @@
+                                        CameraResultFragment cameraResultFragment = new CameraResultFragment();
+                                        Bundle bundle = new Bundle();
+                                        bundle.putParcelable("image", picture);
+                                        cameraResultFragment.setArguments(bundle);
+                                        ((MainActivity) getContext()).getSupportFragmentManager()
+                                                .beginTransaction()
+                                                .replace(R.id.flContainer, cameraResultFragment)
+                                                .addToBackStack(null)
+                                                .commit();
+                                    }
+                                });
+                                Log.i(TAG, "Image saved successfully");
+                            }
+                            @Override
+                            public void onError(ImageCaptureException error) {
+                                Log.e(TAG, "failed");
+                                error.printStackTrace();
+                            }
+                        }
+                );
+            }
+        });
+    }
+
+    private void bindPreview(ProcessCameraProvider cameraProvider) {
+        Preview preview = new Preview.Builder().build();
+        CameraSelector cameraSelector = new CameraSelector.Builder()
+                .requireLensFacing(CameraSelector.LENS_FACING_BACK)
+                .build();
+        imageCapture = new ImageCapture.Builder()
+                .setCaptureMode(ImageCapture.CAPTURE_MODE_MINIMIZE_LATENCY)
+                .setTargetRotation(getActivity().getWindowManager().getDefaultDisplay().getRotation())
+                .build();
+        preview.setSurfaceProvider(previewView.createSurfaceProvider());
+
+        cameraProvider.unbindAll();
+        Camera camera = cameraProvider.bindToLifecycle((LifecycleOwner) this, cameraSelector, preview, imageCapture);
+    }
+
+    private File getPhotoFile() {
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMddHHmmss", Locale.US);
+
+        // Get safe storage directory for photos
+        // getExternalFilesDir used to access package specific directories, so don't need to request external read/write runtime permissions
+        File mediaStorageDir = new File(getContext().getExternalFilesDir(Environment.DIRECTORY_PICTURES), TAG);
+
+        // Create the storage directory if it does not exist
+        if (!mediaStorageDir.exists() && !mediaStorageDir.mkdirs()) {
+            Log.d(TAG, "failed to create directory");
+        }
+
+        // Return the file target for the photo based on filename
+        return new File(mediaStorageDir.getPath() + File.separator + dateFormat.format(new Date()) + ".jpg");
```

Comment:
may be use a Prefix in the Image name like `Image.<timestamp>.jpg` or  `Photofy.<timestamp>.jpg`.

Reply:
I couldn't find how to add a prefix with < > characters, so I made the file name Image_timestamp.jpg

Reply:
I believe Asif meant to say use '.' as a separator.
e.g. image.2239929992.jpg
Not a blocker.